pylana/filter.py

```diff
@@ -0,0 +1,193 @@
+import time
+from dateutil.parser import parse
+
+def combine_filters(*filters) -> list:
+    """
+    Create a list of combined filters.
+
+    Args:
+        *filters:
+            Filter arguments to be combined into a trace filter sequence. They
+            can be either a single filter as a dictionary or multiple filters in
+            a list.
+
+    Returns:
+        A list containing the combined filters.
+    """
+    combined_filter = []
+
+    for trace_filter in filters:
+        if type(trace_filter) is list:
+            combined_filter += trace_filter
+        else:
+            combined_filter.append(trace_filter)
+
+    return(combined_filter)
+
+
+def create_timespan_filter(start: str, end: str) -> dict:
+    """
+    Create a timespan filter to be used in a trace filter sequence.
+
+    Args:
+        start:
+            A string denoting the start timestamp.
+        end:
+            A string denoting the end timestamp.
+
+    Returns:
+        A dictionary containing the filter.
+    """
+    timespan_filter = {'startInRange': False,
+                       'endInRange': False,
+                       'type': 'timeRangeFilter',
+                       'from': time.mktime(parse(start).timetuple()) * 1000,
+                       'to': time.mktime(parse(end).timetuple()) * 1000}
+
+    return(timespan_filter)
+
+
+def create_attribute_filter(attribute_name: str, values: list) -> dict:
+    """
+    Create a categorical attribute filter to be used in a trace filter sequence.
+
+    Args:
+        attribute_name:
+            A string denoting the name of the attribute.
+        values:
+            A list of values to be filtered.
+
+    Returns:
+        A dictionary containing the filter.
+    """
+    attribute_filter = {
+        'type': 'attributeFilter',
+        'attributeName': attribute_name,
+        'values': values
+        }
+
+    return(attribute_filter)
+
+
+def create_numeric_attribute_filter(attribute_name: str, value_min, value_max) -> dict:
+    """
+    Create a numeric attribute filter to be used in a trace filter sequence.
+
+    Args:
+        attribute_name:
+            A string denoting the name of the attribute.
+        value_min:
+            An integer or float denoting the minimum value.
+        value_max:
+            An integer or float denoting the maximum value.
+
+    Returns:
+        A dictionary containing the filter.
+    """
+    numeric_attribute_filter = {
+        'type': 'numericAttributeFilter',
+        'attributeName': attribute_name,
+        'min': value_min,
+        'max': value_max
+        }
+
+    return(numeric_attribute_filter)
+
+
+def create_variant_slider_filter(group_min: int, group_max: int) -> dict:
```

Comment:
we should put the `group` either in the function name or out of the arguments. It should be in the doc-string.

Reply:
done